new-packages/ts-project/src/state.ts

```diff
@@ -268,6 +286,65 @@ export function extractState(
                     'description',
                   );
 
+                  const guardV4 = findProperty(ctx, ts, element, 'cond');
+                  const guardV5 = findProperty(ctx, ts, element, 'guard');
+
+                  if (guardV4 && guardV5) {
+                    ctx.errors.push({
+                      type: 'property_mixed',
+                    });
+                  }
+
+                  let guardBlock;
+                  const guard = guardV5 ?? guardV4;
+
+                  if (guard) {
+                    if (ts.isStringLiteralLike(guard.initializer)) {
+                      guardBlock = createGuardBlock({
+                        sourceId: guard.initializer.text,
+                        parentId: node.uniqueId,
+                      });
+                    } else if (
+                      ts.isObjectLiteralExpression(guard.initializer)
+                    ) {
+                      const typeProperty = findProperty(
+                        ctx,
+                        ts,
+                        guard.initializer,
+                        'type',
+                      );
+
+                      if (
+                        typeProperty &&
+                        ts.isStringLiteralLike(typeProperty.initializer)
+                      ) {
+                        guardBlock = createGuardBlock({
+                          sourceId: typeProperty.initializer.text,
+                          parentId: node.uniqueId,
+                        });
+                      } else {
+                        ctx.errors.push({
+                          type: 'transition_property_unhandled',
+                        });
```

Comment:
This is where an invalid value for the `type` property will raise an error. We can change it either here or later.